karapace/serialization.py

```diff
@@ -207,7 +207,8 @@ def read_value(schema: TypedSchema, bio: io.BytesIO):
 
 def write_value(schema: TypedSchema, bio: io.BytesIO, value: dict) -> None:
     if schema.schema_type is SchemaType.AVRO:
-        writer = DatumWriter(schema.schema)
+        # Expect that schema is the same as writer used, this is the default.
```

Comment:
I don't understand this comment. This is the writer, no? So this is the schema the writer is using.